src/Symfony/Component/Serializer/Normalizer/BackedEnumNormalizer.php

```diff
@@ -63,7 +63,7 @@ public function denormalize($data, string $type, string $format = null, array $c
 
         try {
             return $type::from($data);
-        } catch (\ValueError $e) {
+        } catch (\ValueError|\TypeError $e) {
             throw new InvalidArgumentException('The data must belong to a backed enumeration of type '.$type);
```

Comment:
This case should probably use `NotNormalizableValueException` even for invalid strings btw.

Reply:
The exception was changed here https://github.com/symfony/symfony/pull/47128

Reply:
Yes this commit is the source of the bug.

Reply:
Yet it was also fixing a bug, please make sure that test added in https://github.com/symfony/symfony/pull/47128 keeps passing and/or that the behavior it fixed remains the expected one.

Reply:
All Serializer-related tests, including those in this commit, pass.